HealthGPT/HealthGPT/HealthDataInterpreter.swift

```diff
@@ -8,73 +8,74 @@
 
 import Foundation
 import Spezi
-import SpeziOpenAI
+import SpeziChat
+import SpeziLLM
+import SpeziLLMOpenAI
 import SpeziSpeechSynthesizer
 
 
-class HealthDataInterpreter: DefaultInitializable, Component, ObservableObject, ObservableObjectProvider {
-    @Dependency var openAIComponent = OpenAIComponent()
-
+@Observable
+class HealthDataInterpreter: DefaultInitializable, Module, EnvironmentAccessible {
+    @ObservationIgnored @Dependency private var llmRunner: LLMRunner
     
-    var querying = false {
-        willSet {
-            _Concurrency.Task { @MainActor in
-                objectWillChange.send()
-            }
-        }
-    }
-    
-    var runningPrompt: [Chat] = [] {
-        willSet {
-            _Concurrency.Task { @MainActor in
-                objectWillChange.send()
-            }
-        }
-        didSet {
-            _Concurrency.Task {
-                if runningPrompt.last?.role == .user {
-                    do {
-                        try await queryOpenAI()
-                    } catch {
-                        print(error)
-                    }
-                }
-            }
-        }
-    }
+    var llm: (any LLMSession)?
+    var systemPrompt = ""
     
+    required init() { }
     
-    required init() {}
-
-
-    func generateMainPrompt() async throws {
+    private func generateSystemPrompt() async throws -> String {
         let healthDataFetcher = HealthDataFetcher()
         let healthData = try await healthDataFetcher.fetchAndProcessHealthData()
-
-        let generator = PromptGenerator(with: healthData)
-        let mainPrompt = generator.buildMainPrompt()
-        runningPrompt = [Chat(role: .system, content: mainPrompt)]
+        return PromptGenerator(with: healthData).buildMainPrompt()
```

Comment:
Is it necessary to distribute the logic into different components here?

Reply:
These components are leftovers from the original prototype. We will want to rewrite this logic with function calling along the same lines as LLMonFHIR.